client/my-sites/comments/comment-list/index.jsx

```diff
@@ -132,18 +146,30 @@ export class CommentList extends Component {
 		} );
 	};
 
-	setCommentStatus = ( commentId, postId, status, options = { showNotice: true } ) => {
+	setCommentStatus = ( commentId, postId, status, options = { isUndo: false, persist: false, showNotice: true } ) => {
+		const { isUndo, persist, showNotice } = options;
+
 		// TODO: Replace with Redux getComment()
 		const comment = this.getComment( commentId );
 
 		if ( comment && status === comment.status ) {
 			return;
 		}
 
+		if ( persist ) {
+			this.updatePersistedComments( {
+				...comment,
+				i_like: 'approved' !== status ? false : comment.i_like,
+				status,
+			}, isUndo );
```

Comment:
why can't we just store the ids of the comments that have been approved/unapproved and manually grab them from state in addition to the list of comments the view wants?